migrations/lib/importer/steps/base/site_settings.rb

```diff
@@ -0,0 +1,231 @@
+# frozen_string_literal: true
+
+module Migrations::Importer::Steps::Base
+  class SiteSettings < ::Migrations::Importer::Step
+    class << self
+      def inherited(klass)
+        super
+        klass.requires_mapping :last_change_dates, "SELECT name, updated_at FROM site_settings"
+      end
+    end
+
+    DATATYPES_WITH_DEPENDENCY =
+      [
+        Enums::SiteSettingDatatype::CATEGORY,
+        Enums::SiteSettingDatatype::CATEGORY_LIST,
+        Enums::SiteSettingDatatype::EMOJI_LIST,
+        Enums::SiteSettingDatatype::GROUP,
+        Enums::SiteSettingDatatype::GROUP_LIST,
+        Enums::SiteSettingDatatype::TAG_GROUP_LIST,
+        Enums::SiteSettingDatatype::TAG_LIST,
+        Enums::SiteSettingDatatype::UPLOAD,
+        Enums::SiteSettingDatatype::UPLOADED_IMAGE_LIST,
+        Enums::SiteSettingDatatype::USERNAME,
+      ].map { |number| ::SiteSettings::TypeSupervisor.types[number].to_s }
+
+    DISALLOWED_SITE_SETTINGS = Set.new([:permalink_normalizations]).freeze
+    LIST_SEPARATOR = "|"
+
+    private_constant :DATATYPES_WITH_DEPENDENCY, :DISALLOWED_SITE_SETTINGS
+
+    def execute
+      super
+
+      @log_messages = []
+      @audit_message = I18n.t("importer.site_setting_log_message")
+      @failed_settings = []
+
+      @settings_index =
+        silence_deprecation_warnings do
+          SiteSetting.all_settings(include_hidden: true)
+        end.index_by { |hash| hash[:setting] }
```

Comment:
Accessing deprecated site settings always prints to the console. We don't want that while we create an index of all settings.